src/main/java/org/openhab/tools/analysis/checkstyle/readme/MarkDownCheck.java

```diff
@@ -0,0 +1,137 @@
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.tools.analysis.checkstyle.readme;
+
+import static org.openhab.tools.analysis.checkstyle.api.CheckConstants.BIN_INCLUDES_PROPERTY_NAME;
+import static org.openhab.tools.analysis.checkstyle.api.CheckConstants.BUILD_PROPERTIES_FILE_NAME;
+import static org.openhab.tools.analysis.checkstyle.api.CheckConstants.MARKDONW_EXTENSION;
+import static org.openhab.tools.analysis.checkstyle.api.CheckConstants.PROPERTIES_EXTENSION;
+import static org.openhab.tools.analysis.checkstyle.api.CheckConstants.README_MD_FILE_NAME;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.commonmark.node.Block;
+import org.commonmark.node.FencedCodeBlock;
+import org.commonmark.node.Heading;
+import org.commonmark.node.IndentedCodeBlock;
+import org.commonmark.node.ListBlock;
+import org.commonmark.node.Node;
+import org.commonmark.parser.Parser;
+import org.eclipse.pde.core.build.IBuild;
+import org.eclipse.pde.core.build.IBuildEntry;
+import org.openhab.tools.analysis.checkstyle.api.AbstractStaticCheck;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.puppycrawl.tools.checkstyle.api.CheckstyleException;
+
+/**
+ * Checks the README.md files for:
+ * <ul>
+ * <li>missing empty lines after headers.
+ * <li>missing empty lines before and after code sections.
+ * <li>missing empty lines before and after lists.
+ * </ul>
+ * 
+ * Checks the build.properties for:
+ * <ul>
+ * <li>the README.MD shouldn't be added in build.properties.
+ * <li>the doc folder shouldn't be added in build.properties.
+ * </ul>
+ * <a href="https://www.eclipse.org/smarthome/documentation/development/bindings/docs.html">Eclipse Smarthome
+ * Documentation Guidelines
+ * info</a>
+ *
+ * @author Erdoan Hadzhiyusein - Initial contribution
+ */
+public class MarkDownCheck extends AbstractStaticCheck {
+    private static final String ADDED_README_FILE_IN_BUILD_PROPERTIES_MSG = "README.MD file must not be added to the bin.includes property";
+    private static final String ADDED_DOC_FOLDER_IN_BUILD_PROPERTIES_MSG = "The doc folder must not be added to the bin.includes property";
+    private static final String README_IO_ERROR_MESSAGE = "An exception was caught during processing mardown readme file {}";
+    private static final String DOC_FOLDER_NAME = "doc";
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    public MarkDownCheck() {
+        setFileExtensions(MARKDONW_EXTENSION, PROPERTIES_EXTENSION);
+    }
+
+    @Override
+    protected void processFiltered(File file, List<String> lines) throws CheckstyleException {
```

Comment:
This method has been changed in Checkstyle 8.1, please update - https://github.com/openhab/static-code-analysis/commit/5ae6ff49c48a5941ded214e1fe6f6a0af8452686